astropy/io/misc/parquet.py

```diff
@@ -0,0 +1,304 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+This package contains functions for reading and writing Parquet
+tables that are not meant to be used directly, but instead are
+available as readers/writers in `astropy.table`.  See
+:ref:`astropy:table_io` for more details.
+"""
+
+import os
+import warnings
+
+import numpy as np
+
+# NOTE: Do not import anything from astropy.table here.
+# https://github.com/astropy/astropy/issues/6604
+from astropy.utils.exceptions import AstropyUserWarning
+
+PARQUET_SIGNATURE = b'PAR1'
+
+__all__ = ['read_table_parquet', 'write_table_parquet']
+
+
+def is_parquet(origin, filepath, fileobj, *args, **kwargs):
+
+    if fileobj is not None:
+        pos = fileobj.tell()
+        signature = fileobj.read(4)
+        fileobj.seek(pos)
+
+        return signature == PARQUET_SIGNATURE
+    elif filepath is not None:
+        return filepath.endswith(('.parquet'))
+    else:
+        return False
+
+
+def read_table_parquet(input, columns=None, schema_only=False, filters=None):
+    """
+    Read a Table object from a Parquet file.
+
+    This requires `pyarrow <https://arrow.apache.org/docs/python/>`_
+    to be installed.
+
+    Parameters
+    ----------
+    input : `str` or file-like object
```

Comment:
It would be nice to support any Path-like.
For an example, see 
https://github.com/astropy/astropy/blob/9081f11c21381e8b08ac8d74c46f64296e469246/astropy/cosmology/tests/conftest.py#L37-L42

Reply:
With numpydoc the quotes are not needed for python builtins

```suggestion
    input : str or file-like object
```

Reply:
With numpydoc the quotes are not needed for python builtins

```suggestion
    input : str or file-like object
```